### PR TITLE
feat: Enable Rule CA1822

### PR DESCRIPTION
**Pull request by workgroupengineering**
## What does the pull request do?

[CA1822](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822): Mark members as static

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues

Part of #8944

### CODE REVIEW

File: src/Avalonia.Base/Controls/Classes.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ public Classes(params string[] items)
         /// </remarks>
         public override void Add(string name)
         {
-            ThrowIfPseudoclass(name, "added");
+            Classes.ThrowIfPseudoclass(name, "added");
```
Comment (jp2masa) [inline src/Avalonia.Base/Controls/Classes.cs:66]: Are these `Classes.` really needed? Also on other files, e.g. `KeySpline`, `TaskStreamPlugin`, `DragDropDevice`, ...

File: samples/ControlCatalog/Pages/OpenGlPage.xaml.cs [hunk 2]
```diff
@@ -256,7 +256,7 @@ private void CheckError(GlInterface gl)
 
         protected unsafe override void OnOpenGlInit(GlInterface GL, int fb)
         {
-            CheckError(GL);
+            OpenGlPageControl.CheckError(GL);
```
Comment (maxkatz6) [inline samples/ControlCatalog/Pages/OpenGlPage.xaml.cs:259]: Not sure why we need static class name here and in some other places, as it's used in the same scope.

**Comment by robloo** [comment]
I'm strongly against this one. It's too many changes and the decision to make this static or not should have been made when the code was first written -- not by another person doing many at once.

This type of thing just shouldn't be done...

**Comment by workgroupengineering** [comment]
> This type of thing just shouldn't be done... it changes the API surface in ways you may not understand and may not make sense on a case by case basis.

I'm not entirely true, I just changed the private and internal methods.



> I'm strongly against this one. It's too many changes and the decision to make things static or not should have been made when the code was first written -- not by another person doing many at once.

You're right, that's why it's marked as bezzo. Hope the original authors can do the review.


I think version 11 is an unmissable opportunity to make Avalonia even better. After 11 is released as final, we will have to wait a long time before we take an opportunity like this.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024874-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by robloo** [comment]
>> This type of thing just shouldn't be done... it changes the API surface in ways you may not understand and may not make sense on a case by case basis.
>
>I'm not entirely true, I just changed the private and internal methods.

If so, then it doesn't need to be done before the 11.0 API is locked -- it can be done anytime. Actually, I still think it's better to ignore this rule and only update these things when there is a reason on a case-by-case basis.

Fundamentally the rule states what "Can" be made static. It doesn't say what "Should" be made static. That is an important distinction we would have to review on a case-by-case basis. Which really seems like a lot of time honestly and difficult without original authors reviewing their parts of the code. If you can get the original code authors to comment obviously I have no concerns.

> I think version 11 is an unmissable opportunity to make Avalonia even better. After 11 is released as final, we will have to wait a long time before we take an opportunity like this.

I agree in concept we want to make 11 as good as possible. And API changes are much more difficult after that. I also appreciate the time you are taking going through these sorts of issues. However, I would be much more strategic and only change these types of things when there is a reason. Since this isn't public API too, there is even less reason to do it now before 11.0. I would still ignore this rule.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0025013-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0025842-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by maxkatz6** [comment]
@workgroupengineering let's make this rule as a suggestion instead of warning/error. As in the end it won't improve perf any how significantly. But it will cause inconvenience for the contributors.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026198-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026361-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026824-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

File: samples/ControlCatalog/Pages/OpenGlPage.xaml.cs (hunk 2 above)
Comment (maxkatz6) [reply samples/ControlCatalog/Pages/OpenGlPage.xaml.cs:259]: @workgroupengineering have you checked why we need class name here? There quite a lot of changes in this PR already, but this specifically doesn't seem to be needed.

Comment (workgroupengineering) [reply samples/ControlCatalog/Pages/OpenGlPage.xaml.cs:259]: I did, probably when I have merge I made some mistakes.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026957-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**